### PR TITLE
CMake: In case of C++14 support, check whether std::max/min support constexpr

### PR DESCRIPTION
**Pull request by tamiko**
In reference to #4704

@drwells

### CODE REVIEW

File: cmake/checks/check_01_cxx_features.cmake [hunk 1]
```diff
@@ -184,6 +185,19 @@ IF(NOT DEFINED DEAL_II_WITH_CXX14 OR DEAL_II_WITH_CXX14)
       DEAL_II_HAVE_CXX14_MAKE_UNIQUE)
 
     #
+    # This test checks constexpr std::max/min support
```
Comment (tjhei) [inline cmake/checks/check_01_cxx_features.cmake]: can you add a comment what compiler this is needed for?

**Comment by masterleinad** [comment]
This check fails for gcc-4.9.2 which is what we want.

**Comment by drwells** [comment]
Looks good to me, but we should add the comment about GCC 4.9s C++14 support.

**Comment by drwells** [comment]
LGTM 👍

**Pull request merged**